tools/Raven.Migrator/CosmosDB/CosmosDBMigrator.cs

```diff
@@ -0,0 +1,169 @@
+            {
+                _configuration.CollectionsToMigrate = GetCollectionsToMigrate(client, database);
+            }
+
+            await MigrationHelpers.MigrateNoSqlDatabase(
+                _configuration,
+                async (mongoCollectionName, ravenCollectionName, isFirstDocument, streamWriter) =>
+                    await MigrateSingleCollection(client, database, mongoCollectionName, ravenCollectionName, isFirstDocument, streamWriter));
+        }
+
+        private static async Task MigrateSingleCollection(
+            DocumentClient client,
+            Database database,
+            string cosmosCollectionName,
+            string ravenCollectionName,
+            Reference<bool> isFirstDocument,
+            StreamWriter streamWriter)
+        {
+            var collection = UriFactory.CreateDocumentCollectionUri(database.Id, cosmosCollectionName);
+            var options = new FeedOptions
+            {
+                MaxItemCount = 128
+            };
+
+            var query = client.CreateDocumentQuery<ExpandoObject>(collection, options).AsDocumentQuery();
+            var attachmentNumber = 0;
+
+            while (query.HasMoreResults)
+            {
+                var result = await query.ExecuteNextAsync<ExpandoObject>();
+                foreach (var document in result)
+                {
+                    var dictionary = (IDictionary<string, object>)document;
+                    var documentId = dictionary[CosmosDocumentId].ToString();
+
+                    const string selfLinkKey = "_self";
+                    var selfLink = dictionary[selfLinkKey].ToString();
+
+                    var attachments = new List<Dictionary<string, object>>();
+                    foreach (var attachment in client.CreateAttachmentQuery(selfLink))
+                    {
+                        var attachmentUri = UriFactory.CreateAttachmentUri(database.Id, cosmosCollectionName, documentId, attachment.Id);
+                        var attachmentResponse = await client.ReadAttachmentAsync(attachmentUri);
+                        var resourceMediaLink = attachmentResponse.Resource.MediaLink;
+                        var mediaLinkResponse = await client.ReadMediaAsync(resourceMediaLink);
+
+                        attachmentNumber++;
+
+                        var attachmentInfo = await MigrationHelpers.WriteAttachment(
+                            mediaLinkResponse.Media, mediaLinkResponse.Media.Length,
+                            attachmentResponse.Resource.Id, ravenCollectionName,
+                            mediaLinkResponse.ContentType,attachmentNumber, isFirstDocument, streamWriter);
+
+                        attachments.Add(attachmentInfo);
+                    }
+
+                    await MigrationHelpers.WriteDocument(
+                        document,
+                        documentId,
+                        ravenCollectionName,
+                        new List<string>
```

Comment:
You're generating this const list per document, it would be better to define this as read-only static list

Reply:
done